### ISSUE TITLE
Removal of stale source strings still leaves references to variants

### ISSUE THREAD

**Issue by burner1024** [WeblateOrg/weblate#7637]
### Describe the issue

#7619 is still there.
Added a variant, deleted it, now [source string](https://hive.bgforge.net/translate/fallout/fo2tweaks/en/?checksum=bfd63f4c6a1c901e&q=has%3Avariant+AND+language%3Aen&sort_by=-priority%2Cposition) is a variant of nothing.

Also, any tips on cleaning the tainted strings now?

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. Follow the link.
2. See variant without variants.

### Expected behavior

Variants cleaned up after deletion.

### How do you run Weblate?

Docker container

### Weblate versions

* Weblate: 4.12.2
 * Django: 4.0.4
 * siphashc: 2.1
 * translate-toolkit: 3.6.2
 * lxml: 4.6.5
 * Pillow: 9.1.0
 * bleach: 5.0.0
 * python-dateutil: 2.8.2
 * social-auth-core: 4.2.0
 * social-auth-app-django: 5.0.0
 * django-crispy-forms: 1.14.0
 * oauthlib: 3.2.0
 * django-compressor: 4.0
 * djangorestframework: 3.13.1
 * django-filter: 21.1
 * django-appconf: 1.0.5
 * user-agents: 2.2.0
 * filelock: 3.6.0
 * jellyfish: 0.9.0
 * openpyxl: 3.0.9
 * celery: 5.2.6
 * kombu: 5.2.4
 * translation-finder: 2.12
 * weblate-language-data: 2022.2
 * html2text: 2020.1.16
 * pycairo: 1.21.0
 * pygobject: 3.42.1
 * diff-match-patch: 20200713
 * requests: 2.27.1
 * django-redis: 5.2.0
 * hiredis: 2.0.0
 * sentry_sdk: 1.5.12
 * Cython: 0.29.28
 * misaka: 2.1.1
 * GitPython: 3.1.27
 * borgbackup: 1.2.0
 * pyparsing: 3.0.9
 * pyahocorasick: 1.4.4
 * python-redis-lock: 3.7.0
 * charset-normalizer: 2.0.12
 * Python: 3.10.4
 * Git: 2.30.2
 * psycopg2: 2.9.3
 * psycopg2-binary: 2.9.3
 * phply: 1.2.5
 * ruamel.yaml: 0.17.21
 * tesserocr: 2.5.2
 * akismet: 1.1
 * boto3: 1.22.11
 * zeep: 4.1.0
 * aeidon: 1.11
 * iniparse: 0.5
 * mysqlclient: 2.1.0
 * Mercurial: 6.1.2
 * git-svn: 2.30.2
 * git-review: 2.3.1
 * Redis server: 6.2.7
 * PostgreSQL server: 14.2
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: email-smtp.us-east-1.amazonaws.com
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 4.15.0-176-generic (x86_64)

### Weblate deploy checks

```shell
System check identified some issues:

WARNINGS:
?: (security.W018) You should not have DEBUG set to True in deployment.

System check identified 1 issue (2 silenced).
```

**Comment by burner1024** [comment]
Not sure if I'm reading this right
https://github.com/WeblateOrg/weblate/blob/07c13f9bd6e37832c4c8ae1fa6777e9755c1ad5e/weblate/trans/tasks.py#L184-L186
Source units do have a POT file present, naturally. Could this be reason why it's not working?

**Comment by nijel** [comment]
And what's in the `trans_variant_defining_units` table for matching `trans_variant`?

**Comment by burner1024** [comment]
Not sure I understand the question. This is the whole of both tables.


```
weblate=# select * from trans_variant_defining_units;
 id | variant_id | unit_id 
----+------------+---------
  7 |          1 | 1712767
 12 |          4 | 1711017
 13 |          5 | 1711019
 14 |          6 | 1711021
(4 rows)
```

```
weblate=# select * from trans_variant;
 id | variant_regex |                                                                                                                                 key                                                                                                                                  | component_id 
----+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------
  1 |               |  is already healthy.                                                                                                                                                                                                                                                 |            5
  2 |               | You receive 9500 experience points, because you're a cheating little bastard.                                                                                                                                                                                        |            5
  3 |               | Disabilities healed.                                                                                                                                                                                                                                                 |            5
  4 |               | Are you elder?                                                                                                                                                                                                                                                       |            3
  5 |               | Here is sacrifice for goddes of earth! [Villager throw that stone into well and after short scream remain ethernal silence.] Thank you my friend, now ghost of earth is happy and dont scream anymore. Here is your revard, wonderfull flower which grown near well. |            3
  6 |               | I cant believe! You pass trough temple, nobody made it before! You are now our champion!. Go and talk with elder, she have something very important to tell you.                                                                                                     |            3
  7 |               | Nah... I think I'm gonna take the ladder.                                                                                                                                                                                                                            |            5
(7 rows)
```

**Comment by nijel** [comment]
Thanks, so it really should be fixed by the cleanup in 07c13f9bd6e37832c4c8ae1fa6777e9755c1ad5e, it just was not yet triggered in your case...

**Comment by burner1024** [comment]
Cleanup as in scheduled task? It's been 4 days, should've run by now, no?
